Add builders page.

byte-hedgy opened the pull request:
## Description

1. I created `builders/page.tsx` to display information about checked-in builders.
2. Added a **Builders** link in the `head` navigation.

Regarding the issue in **#3**, the problem might have been caused by the default `blocksBatchSize` value of `500`, which led the browser to continuously send requests without showing any results. One of the requests triggered a "too many requests" error, which may have caused repeated retries.

I’ve increased `blocksBatchSize` to `5000000`, and now everything works as expected.

Screenshots below:
<img width="1987" height="1043" alt="图片" src="https://github.com/user-attachments/assets/e2a1eb74-a4ab-4627-a8a1-26688f6aa6a8" />

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #3_

Your ENS/address: 0xADAeE26D04f78D24F2E73D87944f83E2eE50d6BB

melanke commented on packages/nextjs/components/builders/BuildersTable.tsx:
```diff
@@ -0,0 +1,70 @@
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
+
+type Props = {
+  builders: string[];
```
This parameter name is confusing. It shouldn’t represent all builders, only those who have a profile page.

melanke commented on packages/nextjs/components/builders/BuildersTable.tsx:
```diff
@@ -0,0 +1,70 @@
+          <th>Contract Address</th>
+          <th>Personal Profile</th>
+        </tr>
+      </thead>
+      <tbody>
+        {!filteredEvents || filteredEvents.length === 0 ? (
+          <tr>
+            <td colSpan={3} className="text-center">
+              No Data
+            </td>
+          </tr>
+        ) : (
+          filteredEvents?.map((event, index) => {
+            return (
+              <tr key={index}>
+                <td>
+                  <Address address={event.args.builder} format="short" onlyEnsOrAddress />
+                </td>
+                <td>
+                  <Address address={event.args.checkInContract} format="short" onlyEnsOrAddress />
+                </td>
+                <td>
+                  {builders.includes(event.args.builder ?? "") ? (
+                    <Link
+                      href={`builders/${event.args.builder}`}
+                      target="_blank"
+                      className="text-base text-blue-600 underline hover:text-blue-800 cursor-pointer whitespace-nowrap"
+                    >
+                      🧑‍💻 builders/{event.args.builder}
+                    </Link>
+                  ) : (
+                    "N/A"
+                  )}
+                </td>
+              </tr>
+            );
+          })
+        )}
+      </tbody>
+    </table>
```
Is this table good on mobile devices?

byte-hedgy commented:
hi, @melanke 

Thank you for your review. I have made the following changes based on your feedback:
1. Renamed the parameter `builders` to `publishedBuilders`.
2. Changed to a card grid layout, which is also responsive and works on mobile devices.

Thanks again for taking the time to review!

Screenshots are attached below.

<img width="1668" height="1003" alt="图片" src="https://github.com/user-attachments/assets/75e73e64-8346-4112-bb4e-a3fffa8f3b85" />
<img width="374" height="816" alt="图片" src="https://github.com/user-attachments/assets/54b4f99a-14c9-4425-9edd-7e3b54d7c039" />
